src/Orleans/Runtime/OutsideRuntimeClient.cs

```diff
@@ -237,7 +237,7 @@ private void UnhandledException(ISchedulingContext context, Exception exception)
 
         public async Task Start()
         {
-            await StartInternal().ConfigureAwait(false);
+            await Task.Run(this.StartInternal).ConfigureAwait(false);
```

Comment:
Want to move the original comment from GrainClient.InternalInitialize?
// We deliberately want to run this initialization code on .NET thread pool thread to escape any
// TPL execution environment and avoid any conflicts with client's synchronization context